core/src/main/java/org/apache/iceberg/MetricsModes.java

```diff
@@ -150,6 +150,19 @@ public String toString() {
     }
   }
 
+  /**
+   * Auto promote sorted columns to truncate(16) if default is set at Counts or None.
+   * @param defaultMode default mode
+   * @return mode to use
+   */
+  public static MetricsMode promoteSortedColumnDefault(MetricsMode defaultMode) {
```

Comment:
I'd probably move this into `MetricsConfig` as a private method. Seems like we only use it there.

Reply:
Done